core/src/main/java/org/apache/metamodel/util/FileResource.java

```diff
@@ -19,16 +19,43 @@
 package org.apache.metamodel.util;
 
 import java.io.File;
+import java.io.FileFilter;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Comparator;
 
 /**
  * {@link File} based {@link Resource} implementation.
  */
 public class FileResource implements Resource, Serializable {
 
+    private class DirectoryInputStream extends AbstractDirectoryInputStream<File> {
```

Comment:
Hehe very nitty gritty, but since this nested class is non-static, shouldn't it be declared after static fields?